### ISSUE TITLE
dockerize dashboard

### ISSUE THREAD

**Issue by maxheld83** [subugoe/hoad#29]
... maybe.
would extend #28 and depends on #25

**Comment by maxheld83** [comment]
Contra initial plans as per #29 I've opted **against** standardising on one of the [rocker project](http://rocker-project.org) images as a computing environment for hoad.

Some of these images conveniently include [RStudio Server Open Source](https://rstudio.com/products/rstudio/), which lets you quickly run the IDE in your web browser.
(This approach is also what is powering [binder support in R](https://karthik.github.io/holepunch/)).
This would be convenient for development, but carries some drawbacks for a production-ready data product as hoad inspires to be:

1. The computing environment and the IDE are two separate concerns; they should not be mixed.
2. Actually getting RStudio Server Open Source ready for development is quite complicated (git might need to work inside the container, ...).
3. The rocker project is based on debian images, which enjoy somewhat less support for R binary packages.
4. The package at least (if not the vignette production) need to work on many machines anyway.

**Comment by maxheld83** [comment]
the goal for this would now be to eventually ship a pretty slim container that lets you run the dashboard anywhere.

**Comment by maxheld83** [comment]
so, no we're actually back to `rocker/rstudio:ubuntu`, which conveniently works with RSPM *and* includes RStudio.
I'm going to try and deploy that via azure.

**Comment by maxheld83** [comment]
@njahn82 @jhoeffler just FYI, why I've bumped this up (and am currently working on it):

Deploying to shinyapps.io ran, as we have done so far, has two foundational limitations, one of which I ran into and decided not to invest (waste) more time on debugging it:

1. **No reproducible environment**. Deployment to shinyapps.io happens via `rsconnect::deployApp()`, which in turn relies on some dark {packrat}/{renv} magic.
  The compute environment on shinyapps.io is somewhat undocumented / irreproducible (which R version? which ubuntu? which dependencies?), and `rsconnect::deployApp()` kind of throws out all the work we've put into our reproducible environment.
  The immediate problem was that I couldn't get {googlesheets4} to work on shinyapps.io for some reason probably having to do with build-time dependencies, or whatever.
  The short of it is that, while it works for simpler projects / proofs of concept, shinyapps.io *can* break at any time.
2. There is no good way to use/store secrets in shinyapps.io (such as for using some API), i.e. no facility to store encrypted secrets, as is common for other PaaS/SaaS products. 
  The only way would be to include secrets in plain the source code.
  The code should, in principle, be available only to us (= people with privileges on shinyapps.io), but is in general not recommended to store secrets in unencrypted form.

2. wasn't a problem yet.
1. was a problem right now, so I'm now looking at other deployment options.
